src/hotspot/share/gc/z/zDirector.cpp

```diff
@@ -488,7 +488,8 @@ static bool rule_major_allocation_rate(const ZDirectorStats& stats) {
 
   // Calculate the GC cost for each reclaimed byte
   const double current_young_gc_time_per_bytes_freed = double(young_gc_time) / double(reclaimed_per_young_gc);
-  const double current_old_gc_time_per_bytes_freed = double(old_gc_time) / double(reclaimed_per_old_gc);
+  const double current_old_gc_time_per_bytes_freed = ((reclaimed_per_old_gc == 0) ? (std::numeric_limits<double>::infinity())
+                                                                                  : (double(old_gc_time) / double(reclaimed_per_old_gc)));
```

Comment:
```suggestion
  const double current_old_gc_time_per_bytes_freed = reclaimed_per_old_gc == 0 ? std::numeric_limits<double>::infinity() : (double(old_gc_time) / double(reclaimed_per_old_gc));
```

Sorry I probably confused things here. I think this is what was wanted. I just added all the parentheses as a clarification of how this was meant to be parsed by the compiler.